### PR TITLE
Fix O from explicit.js, add unit tests.

### PR DESCRIPTION
**Pull request by foxdonut**
@barneycarroll as discussed on gitter.

### CODE REVIEW

File: explicit.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ function P(target){
   for(var i = 1; i < arguments.length; i++)
     for(var key in arguments[i])
       if(arguments[i].hasOwnProperty(key))
-        arguments[i][key] === D
+        (arguments[i][key] === D || arguments[i][key] === O)
```
Comment (barneycarroll) [inline explicit.js]: The second expression isn't desirable, the intent is that `O` is always used as a function (and returns `D` in this case).

File: tests/immutable.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ const O = require('../immutable.js')
 const I = x => x
 const A = f => x => f(x)
 
-o('`O` (with a single function argument)', () => {
+o('`O` immutable (with a single function argument)', () => {
```
Comment (barneycarroll) [inline tests/immutable.js]: Nice, that does make the tests read a bit better!

File: explicit.js (hunk 1 above)
Comment (barneycarroll) [reply explicit.js]: [eg](https://flems.io/#0=N4IgZglgNgpgziAXAbVAOwIYFsZJABwwBcBjACwjQGsB7AOgCsEAaEEmtImTvMAVzQkiEDgAIACgAoiGAE4BzGEQCUwADppRosDVmSAbnNERRAXlEBGANzHRAHlFz5fHJzh1YaeUTI2IAan9lDS0tHT1DWVEqGABPY00nF24iOGQIAF1gzVCtCDBJJNdU9Iy6Mgw4AHkAdzRxWRp8GFkiWMkY2OVs3NzChWS3UuROjLNTcwARHt6tAH5RABMYWC5RGQUlEbiMkNnRRHWnLdGzPf3HAeK0zO3Ysco4GUEYGjBRAGVz-YWilJuMncyhh8PgoO0NooiECZhdDn8hrdRho9rIlHxZJpIUoNABfFFofiCYRiD6SEhQGhwDEwVR7fKSACE0gocASTwwLzen263zRRAxmjQMBqn3JlOpaOyex8EHcILB8XMRKEIjQkmWkDQEBJaDpOS0-MFogpVJpGpgWp1apm+LQdo0Kt1EjJ2KIzAS+D4KnUOSNmNEwtFZO+CNSHm43jI43MACZvgsnWqLVbdfrZv7NFI3R7NZRrRwPZQvSpvnbcockxwU-m076M+iA1I89rdR63bDROXRNkHYSBKqxJNJKo+1XNFVJAAPdOiTOXZzXCNeHwJyz2BeDcOeKPfeYSOgK8GSNB8KBQD1huCdrSHNrNblTmOiADk45fe9CCyDYpnn9vLrTp2hyTHiBJELI8T1qIWA0IsZ4wHQMBTvguipGYojAOIhziB6HyHB8HriPhLoepMIEelUhxVHauImsQ5CSLSmG4iArBwCsMCDmgCCICAAAsACciAWAAzCAuLMOg2C4HxjAsGwHBcDwfH4JIeyTt80G5BgByWMwn4AEZ6bGBkGl2Zm5Fpn66dRhRmAAfI4ogALSWMolm9MZdkeWWnnZMobEgBxsDcbxIAWAADIgABsEkZLiQA)

Comment (foxdonut) [reply explicit.js]: Then there is a discrepancy with the README, which has:

```
bish: O
```

also, `O` from `overloaded.js` doesn't work with `O()` to delete.

Comment (barneycarroll) [reply explicit.js]: You're right - sorry!

**Pull request merged**